lib/hdt.js

```diff
@@ -54,20 +54,28 @@ HdtDocumentPrototype.searchTerms = function (options) {
       position = options.position,
       posId = POSITIONS[position],
       prefix = options.prefix || '',
+      subject = options.subject,
       object = options.object;
   if (!(position in POSITIONS))
     return Promise.reject(new Error('Invalid position argument. Expected subject, predicate or object.'));
 
+  if ('subject' in options || 'object' in options) {
+    if (subject === '') return Promise.reject(new Error('Unspecified subject value.'));
+    if (object === '')  return Promise.reject(new Error('Unspecified object value.'));
+    if (posId !== POSITIONS.predicate)  return Promise.reject(new Error('Unsupported position argument. Expected predicate.'));
+  }
+
+  if (subject && object) {
+    // We can call searchTerms directly
```

Comment:
I don't understand this comment. Did you mean `searchTriples`?

Reply:
Yes, you are right. I will change it.